api/controllers/ReferenceController.js

```diff
@@ -53,10 +53,22 @@ module.exports = {
     if (req.params.number && isNaN(req.params.number)) {
       return res.badRequest({err: "Number must be a number"});
     }
+    
+    // Return error when number of given is less than 0
+    // It's a little bit redundant as under normal circumstances this error is never triggered
+    // 0 or bigger value is enforced by using min="0"
+      
+    if (req.params.number < 0) {
+      return res.badRequest({err: "Number must be 0 or more"});
```

Comment:
Sure, but we are checking before the "min='0'" will actually be affected, so this will work.

Reply:
I tested it and in case you have min="0" in the view and enter negative value, "number" is not sent with the request at all. It'll be set to 0. 

I also found another issue that in editing, when you enter negative number the error "number must be 0 or more" is being sent but "there was an issue" is displayed. Therefore I think the best option would be to keep it as it is now.

As previously, please squash on your side.

Reply:
That doesn't make any sense because `min: 0` is on the model, which is only checked when you create or edit a reference. This isn't happening until after this line, so this code must be working, if you send `<0` in a request, it should not be 0 at this point, otherwise there is something else wrong.

Reply:
Oh wait, you put it on the html element as well. That makes sense. Nevermind, ignore me. This check is still good to keep though, just because. If the html doesn't work on a browser, this will catch it instead.